x/rollup/types/adapters_test.go

```diff
@@ -0,0 +1,308 @@
+				ethTxs := make([]hexutil.Bytes, tc.depNum+tc.nonDepNum)
+				transactions := generateEthTransactions(t, tc.depNum, tc.nonDepNum)
+
+				depositTxsNum := 0
+				for i := range transactions {
+					txBinary, err := transactions[i].MarshalBinary()
+					require.NoError(t, err)
+					ethTxs[i] = txBinary
+					if transactions[i].IsDepositTx() {
+						depositTxsNum++
+					}
+				}
+
+				// Convert the binary format to a Cosmos transaction.
+				cosmosTxs, err := rolluptypes.AdaptPayloadTxsToCosmosTxs(ethTxs)
+				require.NoError(t, err)
+
+				if len(ethTxs) == 0 {
+					assert.Equal(t, 0, len(cosmosTxs))
+					return
+				}
+
+				var decodedTx sdktx.Tx
+				err = decodedTx.Unmarshal(cosmosTxs[0])
+				require.NoError(t, err)
+
+				var applyL1TxsRequest rollupv1.ApplyL1TxsRequest
+				err = protoCodec.Unmarshal(decodedTx.GetBody().GetMessages()[0].GetValue(), &applyL1TxsRequest)
+				require.NoError(t, err)
+
+				// Copy the original transaction because time fields are different if not copied.
+				assert.Equal(t, depositTxsNum, len(applyL1TxsRequest.TxBytes))
+				for i, txBytes := range applyL1TxsRequest.TxBytes {
+					newTransaction := transactions[i]
+					err = newTransaction.UnmarshalBinary(txBytes)
+					require.NoError(t, err)
+					assert.Equal(t, transactions[i], newTransaction)
+				}
+
+				for i := 1; i < len(cosmosTxs); i++ {
+					assert.Equal(t, transactions[depositTxsNum-1+i].Data(), []byte(cosmosTxs[i]))
+				}
+			})
+		}
+	})
+
+	t.Run("non-zero txs with error", func(t *testing.T) {
+		t.Run("unmarshal binary error", func(t *testing.T) {
+			cosmosTxs, err := rolluptypes.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{[]byte("invalid")})
+			assert.Nil(t, cosmosTxs)
+			assert.ErrorContains(t, err, "unmarshal binary")
+		})
+		t.Run("zero deposit txs", func(t *testing.T) {
+			inner := generateDynamicFeeInner(r)
+			transaction := ethtypes.NewTx(inner)
+			txBytes, err := transaction.MarshalBinary()
+			require.NoError(t, err)
+			cosmosTxs, err := rolluptypes.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{txBytes})
+			assert.Nil(t, cosmosTxs)
+			assert.Error(t, err)
+		})
+		t.Run("NewAnyWithValue error", func(t *testing.T) {
+			t.Skip()
+			// TODO: Implement this test case
+		})
+		t.Run("depositSDKMsgBytes marshal error", func(t *testing.T) {
+			t.Skip()
+			// TODO: Implement this test case
+		})
+		t.Run("Unpack Cosmos txs error", func(t *testing.T) {
+			depInner := generateDepositInner(r)
+			depTx := ethtypes.NewTx(depInner)
+			depTxBytes, err := depTx.MarshalBinary()
+			require.NoError(t, err)
+
+			nonDepInner := generateDynamicFeeInner(r)
+			nonDepTx := ethtypes.NewTx(nonDepInner)
+			nonDepTxBytes, err := nonDepTx.MarshalBinary()
+			require.NoError(t, err)
+
+			cosmosTxs, err := rolluptypes.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{depTxBytes, nonDepTxBytes, []byte("invalid")})
+			assert.Nil(t, cosmosTxs)
+			assert.ErrorContains(t, err, "unmarshal binary tx: ")
+		})
+	})
+}
```

Comment:
Let's not test the precise error conditions. That makes the tests too dependent on the implementation. The tests should be run from the perspective of someone who only expects a certain behavior and does not know about the implementation. We should be able to refactor the implementation without having to change the tests.